servers/rendering/dummy/storage/texture_storage.h

```diff
@@ -206,6 +206,11 @@ class TextureStorage : public RendererTextureStorage {
 	virtual RID render_target_get_override_velocity(RID p_render_target) const override { return RID(); }
 
 	virtual RID render_target_get_texture(RID p_render_target) override { return RID(); }
+
+	virtual void render_target_set_motion_vectors_target(RID p_render_target, RID p_motion_vectors_texture, RID p_depth_texture, Size2i p_size, int p_view_count) override {}
+	virtual RID render_target_get_motion_vectors_texture(RID p_render_target) const override { return RID(); }
+	virtual RID render_target_get_motion_vectors_depth_texture(RID p_render_target) const override { return RID(); }
+	virtual Size2i render_target_get_motion_vectors_target_size(RID p_render_target) const override { return Size2i(0, 0); }
```

Comment:
I'm personally all for the API given here!

However, I think we need to discuss this among the XR team, because this is side-stepping the existing "velocity texture" on `render_target_set_override()` and `render_target_get_override_velocity()`.

I think those older APIs were designed before we really knew what was required to implement "synthetic frame generation" (which is what Meta's Application Spacewarp is), and should probably just be deprecated. In order to really implement this, we need more than just a single texture, but also a depth texture, and the motion vector target needs to be sized differently than the render target. I don't personally see a use for the older API.

cc @BastiaanOlij @m4gr3d

Reply:
I think this feels like a topic where we need to get the rendering team involved. If what Meta needs out of motion vectors is different from what we currently do with motion vectors, we'll have a problem down the road, especially once we also add this functionality to the Vulkan renderer, or when other vendors enable this for PCVR and we get in the territory where we use motion vector buffers for multiple purposes.

Reply:
Yes, let's do an XR + Rendering maintainers call some time soon! I'll ping on RocketChat at the beginning of next week to try to find a time